pandas 2.0.0rc0: TestPandas.test_masking[False] fails with IntCastingNaNError [Cannot convert non-finite values (NA or inf) to integer]

This started happening in daily cron job where we pull in pre-releases. Any ideas, @taldcroft or @mhvk ?

Example log: https://github.com/astropy/astropy/actions/runs/4228870414/jobs/7344724560

```
Pandas: 2.0.0rc0

________________________ TestPandas.test_masking[False] ________________________

self = <astropy.table.tests.test_table.TestPandas object at 0x7fc718063040>
use_nullable_int = False

    @pytest.mark.parametrize("use_nullable_int", [True, False])
    def test_masking(self, use_nullable_int):
        t = table.Table(masked=True)
    
        t["a"] = [1, 2, 3]
        t["a"].mask = [True, False, True]
    
        t["b"] = [1.0, 2.0, 3.0]
        t["b"].mask = [False, False, True]
    
        t["u"] = ["a", "b", "c"]
        t["u"].mask = [False, True, False]
    
        t["s"] = ["a", "b", "c"]
        t["s"].mask = [False, True, False]
    
        # https://github.com/astropy/astropy/issues/7741
        t["Source"] = [2584290278794471936, 2584290038276303744, 2584288728310999296]
        t["Source"].mask = [False, False, False]
    
        if use_nullable_int:  # Default
            # No warning with the default use_nullable_int=True
            d = t.to_pandas(use_nullable_int=use_nullable_int)
        else:
            with pytest.warns(
                TableReplaceWarning,
                match=r"converted column 'a' from int(32|64) to float64",
            ):
>               d = t.to_pandas(use_nullable_int=use_nullable_int)

astropy/table/tests/test_table.py:2156: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/table/table.py:4004: in to_pandas
    out[name] = Series(out[name], dtype=pd_dtype)
pandas/core/series.py:490: in __init__
    data = sanitize_array(data, index, dtype, copy)
pandas/core/construction.py:582: in sanitize_array
    subarr = _try_cast(data, dtype, copy)
pandas/core/construction.py:769: in _try_cast
    subarr = maybe_cast_to_integer_array(arr, dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

arr = array([nan,  2., nan]), dtype = dtype('int64')

    def maybe_cast_to_integer_array(arr: list | np.ndarray, dtype: np.dtype) -> np.ndarray:
... 
        if is_float_dtype(arr.dtype):
            if not np.isfinite(arr).all():
>               raise IntCastingNaNError(
                    "Cannot convert non-finite values (NA or inf) to integer"
                )
E               pandas.errors.IntCastingNaNError: Cannot convert non-finite values (NA or inf) to integer

../../.tox/py310-test-alldeps-predeps/lib/python3.10/site-packages/pandas/core/dtypes/cast.py:1637: IntCastingNaNError
```

--- a/astropy/table/tests/test_table.py
+++ b/astropy/table/tests/test_table.py
@@ -2151,7 +2151,8 @@ class TestPandas:
         else:
             import pandas
             from packaging.version import Version
-            PANDAS_LT_2_0 = Version(pandas.__version__) < Version('2.0dev')
+
+            PANDAS_LT_2_0 = Version(pandas.__version__) < Version("2.0dev")
             if PANDAS_LT_2_0:
                 with pytest.warns(
                     TableReplaceWarning,
@@ -2160,7 +2161,11 @@ class TestPandas:
                     d = t.to_pandas(use_nullable_int=use_nullable_int)
             else:
                 from pandas.core.dtypes.cast import IntCastingNaNError
-                with pytest.raises(IntCastingNaNError, match=r"Cannot convert non-finite values \(NA or inf\) to integer"):
+
+                with pytest.raises(
+                    IntCastingNaNError,
+                    match=r"Cannot convert non-finite values \(NA or inf\) to integer",
+                ):
                     d = t.to_pandas(use_nullable_int=use_nullable_int)
                 return  # Do not continue
 

--- a/astropy/table/tests/test_table.py
+++ b/astropy/table/tests/test_table.py
@@ -2149,11 +2149,20 @@ class TestPandas:
             # No warning with the default use_nullable_int=True
             d = t.to_pandas(use_nullable_int=use_nullable_int)
         else:
-            with pytest.warns(
-                TableReplaceWarning,
-                match=r"converted column 'a' from int(32|64) to float64",
-            ):
-                d = t.to_pandas(use_nullable_int=use_nullable_int)
+            import pandas
+            from packaging.version import Version
+            PANDAS_LT_2_0 = Version(pandas.__version__) < Version('2.0dev')
+            if PANDAS_LT_2_0:
+                with pytest.warns(
+                    TableReplaceWarning,
+                    match=r"converted column 'a' from int(32|64) to float64",
+                ):
+                    d = t.to_pandas(use_nullable_int=use_nullable_int)
+            else:
+                from pandas.core.dtypes.cast import IntCastingNaNError
+                with pytest.raises(IntCastingNaNError, match=r"Cannot convert non-finite values \(NA or inf\) to integer"):
+                    d = t.to_pandas(use_nullable_int=use_nullable_int)
+                return  # Do not continue
 
         t2 = table.Table.from_pandas(d)
 

